src/components/GalaxyModal.js

```diff
@@ -64,7 +64,7 @@ export const GalaxyModalBase = withDisplayName('GalaxyModal')(
       Utils.withBusyState(setLoading),
       withErrorReportingInModal('Error deleting galaxy instance', onError)
     )(async () => {
-      await Ajax().Apps.app(app.googleProject, app.appName).delete(attachedDataDisk ? shouldDeleteDisk : false)
+      await Ajax().Apps.app(app.cloudContext?.cloudResource, app.appName).delete(attachedDataDisk ? shouldDeleteDisk : false)
```

Comment:
What happens when you call this method with `undefined` as the first argument?

Reply:
This will take some time to test... not sure how long galaxy envs take to start after being paused.

Reply:
![Screen Shot 2022-08-10 at 4 45 19 PM](https://user-images.githubusercontent.com/11773357/184016692-449e33ba-ee27-443a-abb0-8f6d6efd8326.png)

Reply:
If you are going to allow for this to be `undefined`, you need to ensure that the code does something logical in that case. I don't see that `undefined` is a reasonable option for this argument given that it is used as `project` in `api/google/v1/apps/${project}/${name}`.

Reply:
I'm not sure what would be an appropriate argument, I figure getting that error is better than the whole UI stopping due to a missing property?

Reply:
I think you need to make sure you don't attempt to call `Ajax().Apps.app(app.cloudContext?.cloudResource, app.appName).delete` when `app.cloudContext` is undefined.

Reply:
> This will take some time to test... not sure how long galaxy envs take to start after being paused.

May want to mock the Ajax response to be able to iterate faster since Galaxy-related operations tend to take a long time in the backend.

Reply:
> I'm not sure what would be an appropriate argument, I figure getting that error is better than the whole UI stopping due to a missing property?

Perhaps but sometimes not (e.g. the UI could do some undesirable stuff, which can go unnoticed because there aren't obvious errors right away). Anytime using the optional chaining operator (`?`), it's worth thinking if we really need to use it or if it can be a code smell (for the reason I mentioned; i.e., wrong things can be performed without noticing soon). Ideally, it should only be used for things outside the UI's control, like API responses. For those cases, if we could do the validation as close to the source (e.g. API response fetching), we could prevent proliferation of `?` downstream.

Reply:
To be clear, `app.cloudContext` can legitimately be undefined. And therefore the code needs to take that into account and prevent the user from calling this Ajax method while it is undefined. That may involve something like disabling the delete button… I haven't looked at the surrounding code to know what would be reasonable. But at the point where the Ajax call is made, you should be confident that `app.cloudContext` is not undefined because you have already checked for that condition.

Reply:
From what I understand, if we have an app (or runtime) cloudContext should never be undefined, and we already have a null check on app for these methods I'm updating.

The main reason I decided to add the optional operator is because there have been inconsistencies on the backend and I wasn't sure that I could "trust" that "cloudResource" would be the correct name.

I don't believe app.cloudContext could ever be undefined - but this change stems from a backend issue where a response was changed.

I'm just hesitant to add extra bloat to the code with a new if statement check when it's already essentially handled.

Reply:
@n8landolt I'm confused… from your screenshot, it seemed to indicate that you did find a condition in which it is undefined. Can you explain more about the screenshot?

Also, `app.cloudContext?.cloudResource` doesn't help you at all if `cloudResource` is not the correct name. What is says is that if `app.cloudContext` is falsey (undefined/null), then return undefined from `app.cloudContext?.cloudResource` instead of blowing up.

Reply:
As far as I know, the property is supposed to be cloudContext.cloudResource returned from the API, as it stands it is the correct property